### PR TITLE
add setting --tmp arg as process.env.APPIUM_TMP_DIR

### PR DESCRIPTION
**Pull request by KazuCocoa**
## Proposed changes

related to https://github.com/appium/appium-support/pull/112 and https://github.com/appium/appium/issues/12527

`process.env.APPIUM_TMP_DIR` should be set once in Appium process.
In BaseDriver, `this.opts.tmpDir` refers `process.env.APPIUM_TMP_DIR`, and the assignment should happen before calling `super` here.
https://github.com/appium/appium-base-driver/blob/d214a0d1f2aa3aecd8fd4033460ce37176ca4c87/lib/basedriver/driver.js#L49

---

After this and https://github.com/appium/appium-support/pull/112, `--tmp` flag will be respected in all of Appium modules which call `tempDir` module.

I've tested below patterns in ruby_lib_core test suite adding logs in `tempDir` module:

- Run a single device test
    - Android
        - UIA2
        - Espresso
    - iOS
        - XCUITest
- Run tests in parallel in single Appium server instance
    - Android x 3 emulators
        - UIA2
        - Espresso
    - iOS x 2 simulators
        - XCUITest
    - Android x 3, iOS x 2

The ruby_lib_core include screen recording, pull/push apks like https://github.com/appium/appium-android-driver/pull/538

Then, I ensured Appium log had no `/var/folders/xxxx`. They called `--tmp` arg.

## Types of changes

What types of changes does your code introduce to Appium?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](https://github.com/appium/appium/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla.js.foundation/appium/appium)
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by imurchie — COMMENTED** [review]
Is the value of `tmpDir` ever checked for validity?

**Comment by KazuCocoa** [comment]
Yes, https://github.com/appium/appium/blob/master/lib/config.js#L217-L224 creates the path before the `new AppiumDriver(args)`.
Or is it better to make it safer to call `fs.statSync()`? (Is this method proper?)

**Comment by imurchie** [comment]
`mkdirp` is good.

File: lib/appium.js [hunk 1]
```diff
@@ -158,6 +158,13 @@ const pendingDriversGuard = new AsyncLock();
 
 class AppiumDriver extends BaseDriver {
   constructor (args) {
+    // It is necessary to set `--tmp` here since it should set to
```
Comment (jlipps) [inline lib/appium.js]: `since it should be set`

File: lib/appium.js [hunk 2]
```diff
@@ -158,6 +158,13 @@ const pendingDriversGuard = new AsyncLock();
...
+    // process.env.APPIUM_TMP_DIR once at initial time in Appium lifecycle.
```
Comment (jlipps) [inline lib/appium.js]: `at an initial point in the Appium lifecycle`

File: lib/appium.js [hunk 3]
```diff
@@ -158,6 +158,13 @@ const pendingDriversGuard = new AsyncLock();
...
+    // The process argument will be referred by BaseDriver.
```
Comment (jlipps) [inline lib/appium.js]: `will be referenced by BaseDriver`

**Review by jlipps — APPROVED** [review]

**Pull request merged**